In [1]:
import pandas as pd
import numpy as np


In [2]:
df_CASE_CLIENTS = pd.read_excel("21.10 CASE_CLIENTS.xlsx")
df_CASE_CLIENTS

client_id last_name first_name middle_name  age sex
0     10000041307    З*****     Р*****      О*****   30   F
1     10000133158    И*****     М*****      Ю*****   37   M
2       100003773    Е*****     О*****      С*****   50   M
3     10000535718    Е*****     Ф*****      Ч*****   45   M
4     10000583119    И*****     Ф*****      Ф*****   28   M
...           ...       ...        ...         ...  ...  ..
3706    116396091    Д*****     К*****      Ц*****   44   F
3707  11919548693    Д*****     М*****      Ш*****   54   M
3708  11996903167    З*****     О*****      А*****   48   F
3709     13351107    Д*****     П*****      Л*****   53   M
3710   3004669418    Д*****     П*****      Ю*****   44   M

[3711 rows x 6 columns]

In [3]:
df_CASE_CONTRACTS = pd.read_excel("21.10 CASE_CONTRACTS.xlsx")
df_CASE_CONTRACTS

contract_id     contract_num                  product_name    client_id  \
0        21111219  ТТЕ7227715*****  Страхование путешественников  10161870404   
1        21111381  ТТЕ7227715*****  Страхование путешественников   1017159879   
2        21112353  БАДАМСТЕ55*****  Страхование путешественников  10161883357   
3        21113224  ТТЕ7227715*****  Страхование путешественников    100766461   
4        21113292  ТТЕ7227715*****  Страхование путешественников  10171604758   
...           ...              ...                           ...          ...   
3706  31133782315  ДСЖ-5/1539*****  Страхование путешественников    100649013   
3707  31133836775  ТТЕ7227778*****  Страхование путешественников   1017151314   
3708  31133933241  БТАСЕ275 -*****  Страхование путешественников   1006491844   
3709   3114463583  ТТЕ7227778*****  Страхование путешественников   1006513037   
3710   3114463955  ТТЕ7227715*****  Страхование путешественников  10171539646   

     contract_status     currency_name  duration    country  price  \
0          Действует  Российский рубль        10   Беларусь   1096   
1          Действует  Российский рубль         7  Индонезия   1918   
2          Действует  Российский рубль        10   Беларусь   1096   
3          Действует  Российский рубль        14     Грузия   1534   
4          Действует  Российский рубль         7  Аргентина   1918   
...              ...               ...       ...        ...    ...   
3706       Действует  Российский рубль        15     Турция   2466   
3707       Действует  Российский рубль         7  Индонезия   1918   
3708       Действует  Российский рубль        15     Турция   2466   
3709       Действует  Российский рубль        30     Турция   4932   
3710       Действует  Российский рубль         7  Индонезия   1918   

      insurance_amount  
0              2000000  
1              5000000  
2              2000000  
3              2000000  
4              5000000  
...                ...  
3706           3000000  
3707           5000000  
3708           3000000  
3709           3000000  
3710           5000000  

[3711 rows x 10 columns]

In [29]:
df_CASE_CONTRACTS.client_id.nunique()

3711

In [4]:
df_CASE_LOSSES = pd.read_excel("21.10 CASE_LOSSES.xlsx")
df_CASE_LOSSES

loss_id    client_id                    loss_name  loss_payout_amt
0         1  10000713193  Оказание медицинской помощи           100000
1         2    100014656  Оказание медицинской помощи            50000
2         3  10006545463  Оказание медицинской помощи            30000
3         4  10006611194  Оказание медицинской помощи            70000
4         5   1001091597  Оказание медицинской помощи            60000
5         6   1001198959  Оказание медицинской помощи            50000
6         7    100193151  Оказание медицинской помощи           100000
7         8  10034571331  Оказание медицинской помощи            50000
8         9  10041665730  Оказание медицинской помощи            20000
9        10   1004171531  Оказание медицинской помощи           100000
10       11  10049477653  Оказание медицинской помощи           300000
11       12   1004971034  Оказание медицинской помощи            30000
12       13  10053883633  Оказание медицинской помощи            50000
13       14    100543063  Оказание медицинской помощи            40000
14       15    100639851  Оказание медицинской помощи           150000
15       16    100655800  Оказание медицинской помощи            50000
16       17   1009008749  Оказание медицинской помощи            90000
17       18   1010518468  Оказание медицинской помощи            50000
18       19     10107896  Оказание медицинской помощи            20000
19       20    101095518  Оказание медицинской помощи            50000
20       21  10109591977  Оказание медицинской помощи           100000
21       22   1011049180  Оказание медицинской помощи            50000
22       23   1011175413  Оказание медицинской помощи          1000000
23       24  10111991766  Оказание медицинской помощи          1500000
24       25  10118841571  Оказание медицинской помощи           200000
25       26  10139699711  Оказание медицинской помощи           250000
26       27  10146335363  Оказание медицинской помощи           150000
27       28  10151551961  Оказание медицинской помощи           100000
28       29   1015458577  Оказание медицинской помощи           100000
29       30   1015458839  Оказание медицинской помощи            50000
30       31  10159717564  Оказание медицинской помощи           200000
31       32     10160959  Оказание медицинской помощи            50000
32       33  10164911618  Оказание медицинской помощи            80000
33       34  10166187767  Оказание медицинской помощи            50000
34       35    101664801  Оказание медицинской помощи          1000000
35       36  10168743459  Оказание медицинской помощи           800000
36       37  11163199740  Оказание медицинской помощи           500000
37       38   1860164119  Оказание медицинской помощи           700000
38       39   1000484488  Оказание медицинской помощи          1300000
39       40  10007398134  Оказание медицинской помощи           100000
40       41    100077669  Оказание медицинской помощи           100000
41       42  10010634147  Оказание медицинской помощи            50000
42       43  10011174381  Оказание медицинской помощи            80000
43       44  10144491815  Оказание медицинской помощи            30000
44       45   1000579095  Оказание медицинской помощи            20000

In [7]:
df_CASE_LOSSES.client_id.nunique()

45

In [5]:
df_experiment_group= pd.read_excel("21.10 experiment_group.xlsx")
df_experiment_group

client_id    group   experiment_id
0         10001718  control  policy_pricing
1       1017180395  control  policy_pricing
2       1017178047  control  policy_pricing
3       1017174614  control  policy_pricing
4       1017174071  control  policy_pricing
...            ...      ...             ...
14837  11919548695     test  policy_pricing
14838  11919548696     test  policy_pricing
14839  11996903168     test  policy_pricing
14840  11996903169     test  policy_pricing
14841  11996903170     test  policy_pricing

[14842 rows x 3 columns]

In [46]:
df_experiment_group.group.value_counts()

group
test       6306
control    6043
Name: count, dtype: int64

In [28]:
df_experiment_group.client_id.nunique()

14842

In [27]:
df_total = df_experiment_group[['client_id', 'group']].merge(df_CASE_LOSSES[['currency_name', 'loss_payout_amt']], on='client_id',how = 'left')
df_total


client_id    group  loss_payout_amt
0         10001718  control              NaN
1       1017180395  control              NaN
2       1017178047  control              NaN
3       1017174614  control              NaN
4       1017174071  control              NaN
...            ...      ...              ...
14837  11919548695     test              NaN
14838  11919548696     test              NaN
14839  11996903168     test              NaN
14840  11996903169     test              NaN
14841  11996903170     test              NaN

[14842 rows x 3 columns]

In [47]:
df_total.group.value_counts()

group
test       6306
control    6043
Name: count, dtype: int64

In [33]:
df_fin = df_total.merge(df_CASE_CONTRACTS[['client_id','currency_name', 'price',]], on='client_id',how = 'left')
df_fin   

client_id    group  loss_payout_amt     currency_name   price
0         10001718  control              NaN  Российский рубль  3836.0
1       1017180395  control              NaN  Российский рубль  1918.0
2       1017178047  control              NaN  Российский рубль  3836.0
3       1017174614  control              NaN  Российский рубль  1918.0
4       1017174071  control              NaN  Российский рубль  1096.0
...            ...      ...              ...               ...     ...
14837  11919548695     test              NaN               NaN     NaN
14838  11919548696     test              NaN               NaN     NaN
14839  11996903168     test              NaN               NaN     NaN
14840  11996903169     test              NaN               NaN     NaN
14841  11996903170     test              NaN               NaN     NaN

[14842 rows x 5 columns]

In [34]:
df_fin = df_fin.fillna(value=0)
df_fin

client_id    group  loss_payout_amt     currency_name   price
0         10001718  control              0.0  Российский рубль  3836.0
1       1017180395  control              0.0  Российский рубль  1918.0
2       1017178047  control              0.0  Российский рубль  3836.0
3       1017174614  control              0.0  Российский рубль  1918.0
4       1017174071  control              0.0  Российский рубль  1096.0
...            ...      ...              ...               ...     ...
14837  11919548695     test              0.0                 0     0.0
14838  11919548696     test              0.0                 0     0.0
14839  11996903168     test              0.0                 0     0.0
14840  11996903169     test              0.0                 0     0.0
14841  11996903170     test              0.0                 0     0.0

[14842 rows x 5 columns]

In [35]:
df_fin.currency_name.value_counts()

currency_name
0                   11131
Российский рубль     3591
Доллар США            120
Name: count, dtype: int64

In [36]:
df_fin['price'][df_fin['currency_name'] == 'Доллар США'] = df_fin['price'] * 80

C:\Users\semen\AppData\Local\Temp\ipykernel_1916\1314455865.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fin['price'][df_fin['currency_name'] == 'Доллар США'] = df_fin['price'] * 80


In [41]:
df_CASE_CONTRACTS[df_CASE_CONTRACTS['client_id'] == 10134907714]

contract_id     contract_num                      product_name  \
768    212714915  ТТЕ7227778*****  Страхование путешественников USD   

       client_id contract_status currency_name  duration country  price  \
768  10134907714       Действует    Доллар США        10     США     27   

     insurance_amount  
768             50000

In [39]:
df_fin[df_fin['currency_name'] == 'Доллар США']

client_id    group  loss_payout_amt currency_name     price
185    10134907714  control              0.0    Доллар США    2160.0
214     1018111114  control              0.0    Доллар США    4800.0
380     1014111601  control              0.0    Доллар США    4400.0
751    10013813768  control              0.0    Доллар США    4400.0
775    10119630983  control              0.0    Доллар США    2160.0
...            ...      ...              ...           ...       ...
10466  10047089030     test              0.0    Доллар США    8320.0
11019  10086010617     test              0.0    Доллар США  160000.0
11371  10105147478  control              0.0    Доллар США    9200.0
12129  10117991070        0              0.0    Доллар США   13120.0
14724  10193891479     test              0.0    Доллар США    6560.0

[120 rows x 5 columns]

In [42]:
df_fin = df_fin.drop(columns=['currency_name'])

In [43]:
df_fin['profit'] = df_fin['price'] - df_fin['loss_payout_amt']

In [49]:
df_fin = df_fin[df_fin['group'] != 0]
df_fin.rename(columns={'price': 'revenue'}, inplace=True)
df_fin

client_id    group  loss_payout_amt   price  profit
0         10001718  control              0.0  3836.0  3836.0
1       1017180395  control              0.0  1918.0  1918.0
2       1017178047  control              0.0  3836.0  3836.0
3       1017174614  control              0.0  1918.0  1918.0
4       1017174071  control              0.0  1096.0  1096.0
...            ...      ...              ...     ...     ...
14837  11919548695     test              0.0     0.0     0.0
14838  11919548696     test              0.0     0.0     0.0
14839  11996903168     test              0.0     0.0     0.0
14840  11996903169     test              0.0     0.0     0.0
14841  11996903170     test              0.0     0.0     0.0

[12349 rows x 5 columns]

In [50]:
df_fin.group.value_counts()


group
test       6306
control    6043
Name: count, dtype: int64

In [53]:
df_fin.client_id.nunique()

12349

In [56]:
df_fin.to_excel('df_fin.xlsx', index=False)

In [59]:
df_fin = pd.read_excel("df_fin.xlsx")
df_fin

client_id    group  loss_payout_amt  revenue  profit
0         10001718  control                0     3836    3836
1       1017180395  control                0     1918    1918
2       1017178047  control                0     3836    3836
3       1017174614  control                0     1918    1918
4       1017174071  control                0     1096    1096
...            ...      ...              ...      ...     ...
12344  11919548695     test                0        0       0
12345  11919548696     test                0        0       0
12346  11996903168     test                0        0       0
12347  11996903169     test                0        0       0
12348  11996903170     test                0        0       0

[12349 rows x 5 columns]

In [60]:
group = df_fin.groupby('group').agg({'profit': 'sum'}).reset_index()
group

group   profit
0  control  1717229
1     test  5274866

In [61]:
from scipy.stats import chisquare
test = len(df_fin[df_fin['group'] == 'test'])
control = len(df_fin[df_fin['group'] == 'control'])
total = test + control
observed = [test, control]
expected = [total/2, total/2]
chi = chisquare(observed, f_exp = expected)
test, control, total, chi

(6306,
 6043,
 12349,
 Power_divergenceResult(statistic=5.601182281966151, pvalue=0.017948361469529415))

In [76]:
control_array = mean_control.values

In [65]:
from scipy.stats import ttest_ind
mean_control = df_fin[df_fin['group'] == 'control']['profit']
mean_test = df_fin[df_fin['group'] == 'test']['profit']
ttest = ttest_ind(mean_control, mean_test, equal_var=False)
print(ttest, "Средняя прибыль контрольной группы = ",\
    mean_control.mean(), "Средняя прибыль тестовой группы = ", mean_test.mean())


TtestResult(statistic=-1.1877131508074823, pvalue=0.23496995705237053, df=11995.821785527249) Средняя прибыль контрольной группы =  284.16829389376136 Средняя прибыль тестовой группы =  836.4836663495084


# На уровне значимости 0.05, нет статистически значимой разности в прибыли между контрольной и тестовой группой

###Результат t-теста, который вы получили, состоит из нескольких частей:

statistic (статистика) - это значение t-статистики, которое показывает, насколько отличаются средние значения выборок. Чем больше это значение (по модулю), тем больше отличие между выборками.

pvalue (значение p) - это вероятность получить такие же или более экстремальные результаты, если гипотеза о том, что средние значения выборок равны, верна. Если это значение меньше выбранного уровня значимости (обычно 0.05), это говорит о том, что существует статистически значимое различие между выборками.

df (степени свободы) - это количество степеней свободы в t-распределении, вычисляемое на основе выборок.

Исходя из результатов, полученных вами:

Значение статистики t составляет примерно -1.19, что может указывать на незначительное различие между средними значениями обеих групп.
Значение p равно примерно 0.235, что означает, что на уровне значимости 0.05 нет достаточных доказательств для отвержения нулевой гипотезы (гипотезы о равенстве средних значений обеих групп).
Средняя прибыль контрольной группы составляет примерно 284.17, а тестовой группы - примерно 836.48.
Исходя из значения p, мы не можем однозначно отклонить нулевую гипотезу о равенстве средних значений обеих групп. Однако, проведение более подробного анализа данных и контекста эксперимента может быть полезным для интерпретации этих результатов.